### ISSUE TITLE
2.1.7 PayPal Payflow Pro Error

### ISSUE THREAD

**Issue by cadencelabs-master** [magento/magento2#10023]
### Preconditions

1. Magento 2.1.7
2. Set Payment Method in Stores->Configuration->Sales->Payment Methods->Paypal Payflow Pro (Includes Express Checkout)

### Steps to reproduce

1. Add product to cart
2. Go to checkout
3. Place order using PayPal Payflow Pro

### Expected result

1. Successful Checkout

### Actual result

1. An error occurred on the server. Please try to place the order again.

As far as I can tell, this error can be traced to:

file: vendor/magento/module-paypal/Controller/Transparent/Response.php
line: 83

The code is: 

`$this->transaction->getResponseObject($this->getRequest()->getPostValue());`

However, if you are using Payflow pro with the transparent option, this controller is actually executed in the context of a GET action -- meaning getPostValue() returns nothing

The code should be updated to get the parameters from GET

`$response = $this->transaction->getResponseObject($this->getRequest()->getParams());` 

You can see the final response from paypal is:

302 from payflowlink.paypal.com
Location: yourstore.com/paypal/transparent/response/?SECURETOKEN... (GET request)

**Comment by webtheorydesigns** [comment]
I am having a similar issue with my Magento/paypal pro transaction, but I dont see the order being processed in Orders, do you?

**Comment by cadencelabs-master** [comment]
This issue is related to PayPal Payflow pro not allowing the creation of an order, so it may be unrelated.

**Comment by webtheorydesigns** [comment]
Right, it could be.  The only error I get is "An error occurred on the server. Please try to place the order again."

I am able to see the transaction hit my paypal account, and being authorized, but from what Paypal support says, its waiting on Magento to complete the reference transaction authorization. Not sure if that helps you in any way, just thought id add it.

**Comment by cadencelabs-master** [comment]
Oh yea; then this seems related. It's the same behavior we're getting.

**Comment by webtheorydesigns** [comment]
Interesting... I wonder if its worth rolling back to 2.1.6... Not sure if that would help, but I am running out of ideas.

**Comment by cadencelabs-master** [comment]
Did you try the solution I posted?

**Comment by webtheorydesigns** [comment]
Is this the solution?

The code should be updated to get the parameters from GET

$response = $this->transaction->getResponseObject($this->getRequest()->getParams());

**Comment by cadencelabs-master** [comment]
Yes

**Comment by webtheorydesigns** [comment]
Thanks for this post! True life saver! I am now receiving a different type of error, but I think thats paypal related. ('Verified: 10574-This card authorization verification is not a payment )

FYI - I made the change you suggested in Notepad++ and saved the file with UTF-8 encoding.

**Comment by YevSent** [comment]
Hi, @cadencelabs-master described flow, not an issue, this is the expected behavior. The PayPal should always send response to `paypal/transparent/response` by `POST`, not a `GET`.

You need to configure `Return URL Method` in your [manager.paypal.com](https://manager.paypal.com/) account:
`Service Settings` -> `Hosted Checkout Pages` -> `Set Up` -> `Payment Confirmation` section.
![payment confirmation](https://user-images.githubusercontent.com/2736528/27470367-423b02ac-57fc-11e7-85d0-97b269a213b0.png)

The `Enter Return URL` is not necessary because Magento always sends it.

**Comment by morsalex** [comment]
Hello, maybe is not relevant but in the version 2.1.7 i cannot find the button to disable paypal payflow and credit line. Is there any bug with the version?

**Comment by veloraven** [comment]
According to [contributor guide](http://devdocs.magento.com/guides/v2.1/contributor-guide/contributing.html), tickets without response for two weeks should be closed.
If this issue still reproducible please feel free to create the new one: format new issue according to the [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines): with steps to reproduce, actual result and expected result and specify Magento version.

**Comment by Mistrb** [comment]
I was having the same issue on 2.1.10. I changed the setting in Paypal and it fixed it!

![3821890333](https://user-images.githubusercontent.com/35529457/35586094-d95c9c80-05c7-11e8-8819-f4ac001378b1.png)
